finetuner-workflow/finetuner/evaluator.py

```diff
@@ -0,0 +1,213 @@
+#!/bin/which python3
+import argparse
+from typing import Callable, List
+from torch import Tensor
+import resource
+import pynvml
+import psutil
+import time
+import torch
+from transformers import AutoTokenizer, AutoModelForCausalLM, PreTrainedTokenizer, \
+    AutoConfig
+from transformers.modeling_utils import no_init_weights, PreTrainedModel
+
+try:
+    pynvml.nvmlInit()
+except pynvml.nvml.NVMLError_LibraryNotFound:
+    pynvml = None
+
+device = torch.device("cpu")
+if torch.cuda.is_available():
+    device = torch.device("cuda")
+elif torch.backends.mps.is_available():
+    device = torch.device("mps")
+
+def get_gpu_ram() -> str:
```

Comment:
This should be updated to match the identically-purposed function in `finetuner.py` (which was tweaked a bit independently). It should be easiest to just separate it out from both of them into a `utils.py` file and import it, or to use the equivalent and even further updated [`tensorizer.utils.get_mem_usage` function](https://github.com/coreweave/tensorizer/blob/029982083a1278ed45f87cecbb675571d4343125/tensorizer/utils.py#L46-L109) when that dependency is integrated in.